### PR TITLE
Adding Yourself page

### PR DESCRIPTION
**Pull request by AyeRlS**
Adding a simple Yourself page and routing to it.

### CODE REVIEW

File: src/app/core/app-routing.module.ts [hunk 1]
```diff
@@ -10,6 +10,10 @@ const routes: Routes = [
     path: 'items',
     loadChildren: () => import('../modules/item/item.module').then(m => m.ItemModule)
   },
+  {
+    path: 'yourself',
+    loadChildren: () => import('../modules/yourself/yourself.module').then(m => m.YourselfModule)
+  }
```
Comment (smadbe) [inline src/app/core/app-routing.module.ts:16]: Users are groups and multiple things in yourself will be group related stuffs. So it should be in the group module and probably under the url `groups/me`.

File: src/app/core/components/group-nav/group-nav.component.ts [hunk 2]
```diff
@@ -20,10 +21,12 @@ export class GroupNavComponent {
 
   joinedGroups: GroupData = [];
   managedGroups: GroupData = [];
+  selectedGroup: number;
```
Comment (smadbe) [inline src/app/core/components/group-nav/group-nav.component.ts:24]: "0" for yourself while I expect a selected group (so a group id) is misleading. If you just need it for yourself for now, use a boolean.

File: src/app/core/components/group-nav/group-nav.component.html [hunk 3]
```diff
@@ -5,12 +5,12 @@
...
   </div> -->
-  <!-- <div class="group-self" (click)="onSelectYourself($event, 0)">
+  <div class="group-self" (click)="onSelectYourself()">
     <span class="anonym-group">
       <img src="assets/images/Male.svg" style="width: 3.3333rem; height: 3.3333rem;" />
     </span>
-    <span class="group-subtitle" [ngClass]="{'selected': selectedGroup == 0 ? true : false}">Yourself</span>
-  </div> -->
+    <span class="group-subtitle" [ngClass]="{'selected': selectedGroup == 0}">Yourself</span>
+  </div>
```
Comment (smadbe) [inline src/app/core/components/group-nav/group-nav.component.html:13]: The click and the link selection should be handle 100% in html which now if you current page matches the link.

File: src/app/modules/yourself/pages/yourself/yourself.component.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import { CurrentContentService } from 'src/app/shared/services/current-content.service';
+
+const YourselfBreadcrumbCat = 'Yourself';
+
+@Component({
+  selector: 'alg-yourself',
+  templateUrl: './yourself.component.html',
+  styleUrls: ['./yourself.component.scss'],
+})
+export class YourselfComponent implements OnDestroy {
```
Comment (smadbe) [inline src/app/modules/yourself/pages/yourself/yourself.component.ts:11]: Call it "CurrentUserComponent". "Yourself" is just the current name of the link.

**Review by smadbe — COMMENTED** [review]
I am afraid you will have to redo it for scratch 🙄😕 You should ask when you are not 100% how to do it.

File: src/app/modules/yourself/pages/yourself/yourself.component.ts (hunk 4 above)
Comment (AyeRlS) [reply src/app/modules/yourself/pages/yourself/yourself.component.ts]: Done

File: src/app/core/components/group-nav/group-nav.component.ts (hunk 2 above)
Comment (AyeRlS) [reply src/app/core/components/group-nav/group-nav.component.ts:24]: This variable is used to highlight which group tab has been selected, the variable name makes in the file (the tabs are called group-self, group-joined, and group-managed) but I can see why it can be misleading.

How about renaming it to selectedTab ? Also, this variable isn't currently used in the other element, but according to the design it should be, should I add it in another PR or in this one ?

File: src/app/core/app-routing.module.ts (hunk 1 above)
Comment (AyeRlS) [reply src/app/core/app-routing.module.ts]: I've moved it to the group module and under that url.

File: src/app/core/components/group-nav/group-nav.component.html (hunk 3 above)
Comment (AyeRlS) [reply src/app/core/components/group-nav/group-nav.component.html]: Done

**Review by smadbe — APPROVED** [review]
Looks good now 👍 .
It is unfortunate that 95% of the code of the PR  is garbage (code which has been changed/removed afterwards), which make review more complicated (so possibly missing real error), git history messy (last change on modules will be a removal of something which have been added just before). 
You should have followed the recommendation of really redoing it from scratch... it would have been both cleaner and faster for you. Also, it seems you commit too early without having tested a minimum what you commit :-/.

**Review by smadbe — CHANGES_REQUESTED** [review]
BTW, you still have to merge breaking changes in master.

**Review by smadbe — APPROVED** [review]
